client.go

```diff
@@ -124,12 +124,34 @@ func (c *Client) Users() *UserClient {
 }
 
 func (c *Client) Repositories() *RepositoryClient {
-	u := &RepositoryClient{Client: &Client{
+	r := &RepositoryClient{Client: &Client{
 		baseURL:  c.baseURL,
 		username: c.username,
 		password: c.password,
 	}}
 
-	u.baseURLSuffix = "repositories"
-	return u
+	r.baseURLSuffix = "repositories"
+	return r
+}
+
+func (c *Client) Registries() *RegistryClient {
+	r := &RegistryClient{Client: &Client{
+		baseURL:  c.baseURL,
+		username: c.username,
+		password: c.password,
+	}}
+
+	r.baseURLSuffix = "registries"
```

Comment:
This seems to be a repeating pattern. Can we maybe extract the "copy client with separate base URL" into its own method?

```go
func (c *Client) withBaseURL(suffix string) *Client {
	return &Client{
		baseURL:  c.baseURL,
		username: c.username,
		password: c.password,
		baseURLSuffix: suffix,
	}
}
```

Reply:
I agree with replacing and removing some clutter. Before committing, would you be fine with this?
```go
func (c *Client) withURLSuffix(suffix string) *Client {
	return &Client{
		baseURL:       c.baseURL,
		username:      c.username,
		password:      c.password,
		baseURLSuffix: suffix,
	}
}

func (c *Client) Projects() *ProjectClient {
	return &ProjectClient{c.withURLSuffix("projects")}
}

[...]
```

Reply:
Exactly what I had in mind :+1: